packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -265,6 +273,21 @@ export async function activate(context: vscode.ExtensionContext) {
     scratchOrgDecorator.showOrg();
     scratchOrgDecorator.monitorConfigChanges();
   }
+
+  const api: any = {
+    CancellableStatusBar,
+    CompositeParametersGatherer,
+    SelectFileName,
+    SelectStrictDirPath,
+    SfdxCommandlet,
+    SfdxCommandletExecutor,
+    SfdxWorkspaceChecker,
+    channelService,
```

Comment:
By exporting the singleton services, we ensure that there is only one instance of it so we won't have any weird behavior.